docs/source/what_is_jupyter.md

```diff
@@ -0,0 +1,412 @@
+
+The graph below presents the best known software components of the Project.
+
+Connections and groups in this diagram are not formal relationships, but
+simple indicators to help us draw the big picture in the next sections.
+
+```mermaid
+graph TD
+    IPython
+        click IPython href "https://ipython.org" _blank
+
+    subgraph source/code
+        Nbformat[["Notebook file format (.ipynb)"]]
+            click Nbformat href "https://nbformat.readthedocs.io" _blank
+
+        JupyterWidgets>Jupyter-Widgets]
+            click JupyterWidgets href "http://ipywidgets.readthedocs.io" _blank
+    end
+
+    subgraph frontends
+        direction LR
+
+        JupyterLab(Jupyter-Lab)
+            click JupyterLab href "https://jupyterlab.readthedocs.io" _blank
+
+        JupyterNotebook(Jupyter-Notebook)
+            click JupyterNotebook href "https://jupyter-notebook.readthedocs.io" _blank
+    end
+
+    subgraph viewers
+        direction LR
+
+        Binder(Binder)
+            click Binder href "https://mybinder.org" _blank
+
+        Voila(Voilà)
+            click Voila href "https://voila.readthedocs.io" _blank
+
+        Nbviewer[nbviewer]
+            click Nbviewer href "https://nbviewer.org" _blank
+
+        Nbconvert[nbconvert]
+            click Nbconvert href "https://nbconvert.readthedocs.io" _blank
+    end
+
+    JupyterHub(Jupyter-Hub)
+        click JupyterHub href "https://jupyterhub.readthedocs.io" _blank
+
+    JupyterHub --> frontends
+    IPython ~~~ source/code
+    frontends --> source/code
+    viewers --> source/code
+```
+
+### Notebook Editor Programs
+
+- [JupyterLab](https://jupyterlab.readthedocs.io/en/latest/)
+- [Jupyter Notebook](https://jupyter-notebook.readthedocs.io/en/stable/)
+
+The two most popular notebook editor programs are Jupyter-Notebook and
+JupyterLab, though there are others. Which one you choose to use is
+mostly a matter of preference (both Notebook and Lab provide many of the
+same capabilities for editing and running notebooks):
+
+- Jupyter Notebook offers a simplified, lightweight notebook authoring
+  experience
+
+- JupyterLab offers a more customizable, feature-rich, tabbed multi-notebook
+  editing environment, and has additional tools like a customizable layout
+  and system console (a common tool used by programmers)
+
+- And more...read about additional notebook interfaces here (LINK)!
+
+A professor, for instance, might use JupyterLab for their daily work as a data
+analyst, and Jupyter Notebook while teaching to provide a simpler, more focused
+view of the notebook's content to their students.
+
+### The notebook format
+
+- [nbformat.readthedocs.io](https://nbformat.readthedocs.io)
+
+Jupyter uses the .ipynb format to store your notebook files (where the data
+and outputs of your notebook reside for long term storage). The .ipynb format
+is interpreted and modified by the nbformat software library.
+
+TODO: features, more info, basic exmaplanation of how it works/structure
```

Comment:
Not sure if this is necessary in a basic introduction.

Reply:
So to elaborate on some principles guiding this document:

The primary goal is to clarify and disambiguate essential ideas and components of the Jupyter ecosystem.

The notebook editor programs are used as an organizing pillar in this document because conceptually they're easy to apprehend (given their similarities to common document editor programs) and can serve as a central conceptual hub to build additional concepts and understanding onto.

The notebook format needs to be included because it contributes to confusion around the term "notebook", and because it is a foundational piece underlying Jupyter/the editors. It is included because users should be clear on the difference between Notebook (the Editor), notebook (a notebook file), computational notebooks (the notebook concept based on literate programming) and notebook (the .ipynb file format).

Eliminating confusion around those concepts is a key goal, because these issues form a big part of users' uncertainty about which pieces are there and what they do.

To be clear, this document should be something users can read early on in their Jupyter experience. I have confidence that, with plain language, and clear guidance, our users are capable of digesting each of these pieces.